Makefile

```diff
@@ -16,8 +16,17 @@ clean:
 	@rm -rfv bin coverage.xml
 
 validate: 
-	@echo "=== $(INTEGRATION) === [ validate ]: Validating source code running golangci-lint..."
-	@go run $(GOFLAGS) $(GOLANGCI_LINT) run --verbose
+ifeq ($(strip $(GO_FILES)),)
+	@echo "=== $(INTEGRATION) === [ validate ]: no Go files found. Skipping validation."
+else
+	@printf "=== $(INTEGRATION) === [ validate ]: running golangci-lint & semgrep... "
+	@go run  $(GOFLAGS) $(GOLANGCI_LINT) run --verbose
+	@if [ -f .semgrep.yml ]; then \
+        docker run --rm -v "${PWD}:/src:ro" --workdir /src returntocorp/semgrep -c .semgrep.yml ; \
+    else \
+    	docker run --rm -v "${PWD}:/src:ro" --workdir /src returntocorp/semgrep -c p/golang ; \
+    fi
```

Comment:
Few comments:
- Seems like tabs and spaces are mixed here.
- I would suggest to do the `if` and sotre the arg for `-c` in a variable, and then call docker run with that variable.